corehq/motech/value_source.py

```diff
@@ -278,21 +278,17 @@ def __eq__(self, other):
         )
 
     def deserialize(self, external_value):
-        """
-        Ignores ``external_value`` and returns ``self.value`` cast from
-        ``self.value_data_type`` to ``self.commcare_data_type``.
-        """
-        serializer = (serializers.get((self.value_data_type, self.commcare_data_type))
-                      or serializers.get((None, self.commcare_data_type)))
-        return serializer(self.value) if serializer else self.value
+        return self._get_commcare_value(self.value)
 
     def _get_commcare_value(self, case_trigger_info):
```

Comment:
it looks its only method is exchanged. How is it simplifying the class inheritance?

Reply:
This change is meant to make `ConstantValue` work like its base class `ValueSource`, so when we extend it, or use mix-ins, subclasses don't have to override `serialize()` or `deserialize()`.

As I was writing this comment to explain how it should work, I figured I'd write a test to check I was right. Turns out ... the code was wrong! So I'm very glad you asked. (And I should have written the tests to begin with.)

`ValueSource.serialize()` converts from CommCare data type to external data type, and `ValueSource.deserialize()` converts from external data type to CommCare data type. This change makes that true for `ConstantValue` too. So when we extend it, e.g. with `JsonPathMixin` in PR #25862, the subclass doesn't need to do anything.